### ISSUE TITLE
Flutter beta 3.24.0-0.1.pre - `android_profile_arm64` timed out generating gn files

### ISSUE THREAD

**Issue by christopherfujino** [flutter/flutter#152123]
From https://github.com/flutter/engine/pull/54003/checks?check_run_id=27656678082
Part of https://github.com/flutter/flutter/issues/152118

`Linux mac_android_aot_engine` ([build](https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20mac_android_aot_engine)) -> ci/android_profile_arm64 ([build](https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20Engine%20Drone/1002908/infra)) -> timed out running gn ([logs](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8741995510014572577/+/u/gn_--target-dir_ci_android_profile_arm64_--runtime-mode_profile_--android_--android-cpu_arm64_--rbe_--no-goma_--xcode-symlinks_--rbe-server-address_unix:___Volumes_Work_s_w_ir_x_w_recipe_cleanup_rbe5gwj4o9x_reproxy.sock_--no-lto/stdout), [swarming task](https://chromium-swarm.appspot.com/task?id=6ae3166ae8f6fd10&o=true&w=true))

**Comment by zanderso** [comment]
https://github.com/flutter/buildroot/pull/876 might not fix, but it will at least make the bad state time out faster.

**Comment by christopherfujino** [comment]
> [flutter/buildroot#876](https://github.com/flutter/buildroot/pull/876) might not fix, but it will at least make the bad state time out faster.

Interesting, do those python scripts run as part of the gn invocation?

**Comment by zanderso** [comment]
> Interesting, do those python scripts run as part of the gn invocation?

Yes. GN needs to know the location of the macOS and iOS SDKs in order to compute the sysroot etc. flags to invoke clang with.

**Comment by christopherfujino** [comment]
> > Interesting, do those python scripts run as part of the gn invocation?
> 
> Yes. GN needs to know the location of the macOS and iOS SDKs in order to compute the sysroot etc. flags to invoke clang with.

Cool, sounds like we should do that in the 3.24 branch then. If we need to kick off new builds for this current release, we can CP it. Otherwise, this is something we should do before the next 3.24 release.
